docs/src/modules/components/AppSearch.js

```diff
@@ -60,100 +64,135 @@ function initDocsearch(userLanguage) {
-        opacity: 1,
-        padding: '5.33px 10.66px',
-        textAlign: 'right',
-        width: '25%',
-      },
-      '& .algolia-docsearch-suggestion .algolia-docsearch-suggestion--content': {
-        float: 'right',
-        padding: '5.33px 0 5.33px 10.66px',
-        width: '75%',
-      },
-      '& .algolia-docsearch-suggestion--subcategory-column-text': {
-        color: theme.palette.text.secondary,
-        fontWeight: theme.typography.fontWeightRegular,
-      },
-      '& .algolia-docsearch-suggestion--highlight': {
-        color: theme.palette.type === 'light' ? '#174d8c' : '#acccf1',
-      },
-      '& .algolia-docsearch-suggestion': {
-        textDecoration: 'none',
-        backgroundColor: theme.palette.background.paper,
-      },
-      '& .algolia-docsearch-suggestion--title': {
-        ...theme.typography.h6,
-        color: theme.palette.text.primary,
-      },
-      '& .algolia-docsearch-suggestion--text': {
-        ...theme.typography.body2,
-        color: theme.palette.text.secondary,
+    },
+    root: {
+      fontFamily: theme.typography.fontFamily,
+      position: 'relative',
+      marginRight: theme.spacing(2),
+      marginLeft: theme.spacing(1),
+      borderRadius: theme.shape.borderRadius,
+      backgroundColor: fade(theme.palette.common.white, 0.15),
+      '&:hover': {
+        backgroundColor: fade(theme.palette.common.white, 0.25),
       },
-      '&& .algolia-docsearch-suggestion--no-results': {
-        width: '100%',
-        '&::before': {
-          display: 'none',
+      '& $inputInput': {
+        transition: theme.transitions.create('width'),
+        width: 120,
+        '&:focus': {
+          width: 170,
         },
       },
     },
-  },
-  root: {
-    fontFamily: theme.typography.fontFamily,
-    position: 'relative',
-    marginRight: theme.spacing(2),
-    marginLeft: theme.spacing(1),
-    borderRadius: theme.shape.borderRadius,
-    backgroundColor: fade(theme.palette.common.white, 0.15),
-    '&:hover': {
-      backgroundColor: fade(theme.palette.common.white, 0.25),
+    search: {
+      width: theme.spacing(9),
+      height: '100%',
+      position: 'absolute',
+      pointerEvents: 'none',
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
     },
-    '& $inputInput': {
-      transition: theme.transitions.create('width'),
-      width: 120,
-      '&:focus': {
-        width: 170,
-      },
+    inputRoot: {
+      color: 'inherit',
+    },
+    inputInput: {
+      padding: theme.spacing(1, 1, 1, 9),
     },
-  },
-  search: {
-    width: theme.spacing(9),
-    height: '100%',
-    position: 'absolute',
-    pointerEvents: 'none',
-    display: 'flex',
-    alignItems: 'center',
-    justifyContent: 'center',
-  },
-  inputRoot: {
-    color: 'inherit',
-  },
-  inputInput: {
-    padding: theme.spacing(1, 1, 1, 9),
-  },
-}));
+  }),
+  { name: 'AppSearch' },
```

Comment:
Do we need to provide a name?

Reply:
Better DX. IMO I would add a lint rule against omitting the name. Makes CSS debugging hard.

Reply:
Maybe we need a Babel plugin that adds it automatically based on the filename 🤔

Reply:
Makes sense Probably based on the var name since this will automatically work for multiple components per file.